src/Common/src/System/Net/Logging/GlobalLog.cs

```diff
@@ -173,41 +180,52 @@ public static void Enter(string functionName, string parameters)
             EventSourceLogging.Log.FunctionStart(functionName, parameters);
         }
 
-        public static void Assert(bool condition, string messageFormat, params object[] data)
+        public static void Assert(bool condition, string message)
         {
             if (!condition)
             {
-                string fullMessage = string.Format(CultureInfo.InvariantCulture, messageFormat, data);
-                int pipeIndex = fullMessage.IndexOf('|');
-                if (pipeIndex == -1)
-                {
-                    Assert(fullMessage);
-                }
-                else
-                {
-                    int detailLength = fullMessage.Length - pipeIndex - 1;
-                    Assert(fullMessage.Substring(0, pipeIndex), detailLength > 0 ? fullMessage.Substring(pipeIndex + 1, detailLength) : null);
-                }
+                Assert(message);
             }
         }
 
-        public static void Assert(string message)
+        public static void Assert(bool condition, string messageFormat, params object[] data)
         {
-            Assert(message, null);
+            if (!condition && !IsEnabled)
```

Comment:
FWIW, I think I'd prefer that we check for `IsEnabled` before calling `Assert`, `AssertFormat`, etc. rather than in the bodies of those methods for better consistency with `Print`.